src/test/java/com/google/sps/GameStageDatabaseTest.java

```diff
@@ -0,0 +1,69 @@
+package com.google.sps;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.sps.data.GameStage;
+import com.google.sps.data.GameStageDatabase;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests the GameStageDatabase to ensure that it puts and queries entities correctly */
+@RunWith(JUnit4.class)
+public final class GameStageDatabaseTest {
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private GameStageDatabase gameStageDatabase;
+  private static final String WEBDEV_LEVEL1_NAME = "Web Developer Level 1";
+  private static final String WEBDEV_INPUT = "Web Developer";
+  private static final String LEVEL_1 = "1";
+  private static final String LEVEL_2 = "2";
+  private static final String TEST_CONTENT = "Some test content here.";
+  private static final Boolean IS_NOT_LAST_STAGE = false;
+  private static final Gson gson = new Gson();
+
+  @Before
+  public void setUp() {
+    helper.setUp(); // initialize local datastore for testing
+    DatastoreService localDatastore = DatastoreServiceFactory.getDatastoreService();
+    this.gameStageDatabase = new GameStageDatabase(localDatastore);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+   * Tests that the objects that are put into the database as entities are queried back as the same
+   * objects
+   */
+  @Test
+  public void putGameStageAndQueryForGameStage_AssertThatTheyAreEqual() {
```

Comment:
This should be two tests:

storeGameStage_worksSuccessfully

and

getGameStage_worksSuccessfully.

As written it's the second test. The first test should store in the database and just assert that no exceptions get thrown. That way if, for examnple, getGameStage is broken, the getGameStage test will break and it will be obvious. As you have it, if getGameStage is broken, the test failure doesn't clearly indicate to you what the problem is.

Reply:
+1